src/System.Management.Automation/security/CatalogHelper.cs

```diff
@@ -400,69 +398,62 @@ internal static string ProcessFilePathAttributeInCatalog(IntPtr memberAttrInfo)
         internal static string CalculateFileHash(string filePath, string hashAlgorithm)
         {
             string hashValue = string.Empty;
-            IntPtr catAdmin = IntPtr.Zero;
 
             // To get handle to the hash algorithm to be used to calculate hashes
-            if (!NativeMethods.CryptCATAdminAcquireContext2(ref catAdmin, IntPtr.Zero, hashAlgorithm, IntPtr.Zero, 0))
+            SafeCATAdminHandle catAdmin;
+            try
+            {
+                catAdmin = WinTrustMethods.CryptCATAdminAcquireContext2(hashAlgorithm);
+            }
+            catch (Win32Exception e)
             {
-                ErrorRecord errorRecord = new ErrorRecord(new InvalidOperationException(StringUtil.Format(CatalogStrings.UnableToAcquireHashAlgorithmContext, hashAlgorithm)), "UnableToAcquireHashAlgorithmContext", ErrorCategory.InvalidOperation, null);
+                ErrorRecord errorRecord = new ErrorRecord(
+                    new InvalidOperationException(StringUtil.Format(CatalogStrings.UnableToAcquireHashAlgorithmContext, hashAlgorithm), e),
+                    "UnableToAcquireHashAlgorithmContext",
+                    ErrorCategory.InvalidOperation,
+                    null);
                 _cmdlet.ThrowTerminatingError(errorRecord);
+                return hashValue;  // Satisfy using check below
```

Comment:
This comment appears at multiple places, but it's not clear to me what it actually means. Can you please update it to make it clearer?

Reply:
Sure thing